test/replication-luatest/linearizable_read_test.lua

```diff
@@ -0,0 +1,273 @@
+        })
+        g.proxies[i]:start({force = true})
+    end
+    g.cluster:start()
+    g.cluster:wait_fullmesh()
+    g.servers[1]:exec(function()
+        box.schema.space.create('sync', {is_sync=true})
+        box.space.sync:create_index('pk')
+    end)
+    g.servers[2]:wait_vclock_of(g.servers[1])
+    g.servers[3]:wait_vclock_of(g.servers[1])
+end)
+
+g.after_all(function()
+    g.cluster:drop()
+end)
+
+g.test_wait_others = function(g)
+    g.proxies[1]:pause()
+    local fid = g.servers[1]:exec(function()
+        local t = require('luatest')
+        local f = require('fiber').new(function()
+            return pcall(box.begin, {linearizable = true})
+        end)
+        f:set_joinable(true)
+        t.assert_equals(f:status(), 'suspended', 'begin() waits for replicas')
+        return f:id()
+    end)
+    g.proxies[1]:resume()
+    local ok = g.servers[1]:exec(function(fid)
+        local fiber = require('fiber')
+        local _, ok, err =  fiber.join(fiber.find(fid))
+        return ok, err
+    end, {fid})
+    t.assert(ok, 'begin() succeeds after replica connections are established')
+end
+
+g.test_timeout = function(g)
+    g.servers[1]:exec(function()
+        local t = require('luatest')
+        box.error.injection.set('ERRINJ_RELAY_FROM_TX_DELAY', true)
+        local ok, err = pcall(box.begin, {linearizable = true, timeout = 0.01})
+        box.error.injection.set('ERRINJ_RELAY_FROM_TX_DELAY', false)
+        t.assert(not ok, 'Error when relay is unresponsive')
+        t.assert_equals(err.type, 'TimedOut',
+                        'Timeout when relay is unresponsive')
+    end)
+end
+
+g.before_test('test_no_dirty_reads', function(g)
+    -- Make cluster size 5. Having quorum 3 out of 5 will make the instance
+    -- poll 5 - 3 + 1 == all 3 of the "real" nodes.
+    g.servers[1]:exec(function()
+        local uuid = require('uuid')
+        box.space._cluster:insert{4, tostring(uuid.new())}
+        box.space._cluster:insert{5, tostring(uuid.new())}
+    end)
+    g.servers[2]:wait_vclock_of(g.servers[1])
+    g.replication_synchro_quorum = g.servers[2]:exec(function()
+        box.ctl.promote()
+        local q = box.cfg.replication_synchro_quorum
+        box.cfg{replication_synchro_quorum = 4}
+        return q
+    end)
+    g.servers[1]:wait_vclock_of(g.servers[2])
+    g.servers[3]:wait_vclock_of(g.servers[2])
+end)
+
+g.after_test('test_no_dirty_reads', function(g)
+    g.servers[2]:exec(function(q)
+        box.cfg{replication_synchro_quorum = q}
+        box.ctl.demote()
+    end, {g.replication_synchro_quorum})
+    g.servers[1]:wait_vclock_of(g.servers[2])
+    g.servers[1]:exec(function()
+        box.space._cluster:delete{4}
+        box.space._cluster:delete{5}
+    end)
+    g.servers[2]:wait_vclock_of(g.servers[1])
+    g.servers[3]:wait_vclock_of(g.servers[1])
+end)
+
+g.test_no_dirty_reads = function(g)
+    -- So that server 2 doesn't receive acks and thus doesn't confirm the
```

Comment:
It seems there is a missing sentence before `-- So ...`.

Reply:
I just removed this comment. `replication_synchro_quorum` is made 4 on server2 in `before_test`. So this commen must relate to that. I think it's excess.